feathr-impl/src/main/scala/com/linkedin/feathr/offline/client/FeathrClient.scala

```diff
@@ -68,6 +68,15 @@ class FeathrClient private[offline] (sparkSession: SparkSession, featureGroups:
     SparkFeaturizedDataset(joinedDF, FeaturizedDatasetMetadata(header=Some(header)))
   }
 
+  /**
+   * APU which joins features and also returns any swallowed exception msgs.
+   */
+  @InternalApi
+  def joinFeaturesWithSwallowedExceptions(joinConfig: FeatureJoinConfig, obsData: SparkFeaturizedDataset,
+    jobContext: JoinJobContext = JoinJobContext()): (SparkFeaturizedDataset, String) = {
```

Comment:
I think it would be better to have a return type to be more extensible, like a Map. That would make the API more generic. 

This would be to address any other future use cases where we want to send some other data back to FFE from the internal feathr engine. 

Therefore, we could have an API called `joinFeaturesExtended` OR `joinFeaturesV2` OR `joinFeaturesWithDataReturn`. 

LMK what you think. The issue with this design is that we would have to change it if we decide to send some other data back from the engine.

Reply:
An example of a metatdata map expected to return:
```
{
"features-with-missing-data": <feature-name-to-value-patched Map (to be sent over Iris notifications)>
"features-with-missing-data-exception": <complete exception string (to be logged in Azkaban logs)>
}
```

Reply:
Done.